tests/test_quickbooks_interrupted_sync.py

```diff
@@ -0,0 +1,186 @@
+                        #       equal to the state for streams that were replicated during the interrupted sync.
+                        # - Verify the interrupted sync replicates the expected record set all interrupted records are in full records
+                        for record in interrupted_records:
+                            rec_time = dt.strptime(record.get('MetaData').get('LastUpdatedTime'), "%Y-%m-%dT%H:%M:%S.%fZ")
+                            self.assertGreaterEqual(rec_time, interrupted_stream_datetime)
+
+                            self.assertIn(record, full_records, msg='Incremental table record in interrupted sync not found in full sync')
+
+                        # Record count for all streams of interrupted sync match expectations
+                        full_records_after_interrupted_bookmark = 0
+                        for record in full_records:
+                            rec_time = dt.strptime(record.get('MetaData').get('LastUpdatedTime'), "%Y-%m-%dT%H:%M:%S.%fZ")
+                            if rec_time >= interrupted_stream_datetime:
+                                full_records_after_interrupted_bookmark += 1
+
+                        self.assertEqual(full_records_after_interrupted_bookmark, interrupted_record_count, \
+                                         msg='Expected {} records in each sync'.format(full_records_after_interrupted_bookmark))
+
+                    else:
+                        # Get the date to start 2nd sync for non-interrupted streams
+                        synced_stream_bookmark = state['bookmarks'].get(stream, {}).get('LastUpdatedTime')
+                        if synced_stream_bookmark:
+                            synced_stream_datetime = dt.strptime(self.convert_state_to_utc(synced_stream_bookmark), "%Y-%m-%dT%H:%M:%SZ")
+                        else:
+                            synced_stream_datetime = start_date_datetime
+
+                        # Verify we replicated some records for the non-interrupted streams
+                        self.assertGreater(interrupted_record_count, 0)
+
+                        # - Verify resuming sync only replicates records with replication key values greater or equal to
+                        #       the state for streams that were replicated during the interrupted sync.
+                        # - Verify resuming sync replicates all records that were found in the full sync (non-interupted)
+                        for record in interrupted_records:
+                            rec_time = dt.strptime(record.get('MetaData').get('LastUpdatedTime'), "%Y-%m-%dT%H:%M:%S.%fZ")
+                            self.assertGreaterEqual(rec_time, synced_stream_datetime)
+
+                            self.assertIn(record, full_records, msg='Unexpected record replicated in resuming sync.')
+
+                elif expected_replication_method == self.FULL_TABLE:
```

Comment:
There are no full table streams under test here. There don't appear to be any full table streams in the tap based on the expectations in base.py

Reply:
Updated the code and removed conditions for checking the replication method for expected streams.